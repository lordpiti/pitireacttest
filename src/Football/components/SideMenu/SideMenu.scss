.active, .active li, .active li:hover {
    background-color: black;
    color: white;
}