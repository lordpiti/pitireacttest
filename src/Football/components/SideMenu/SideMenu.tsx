import React, { FunctionComponent } from 'react';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import { withStyles, Theme } from '@material-ui/core/styles';
import { NavLink } from 'react-router-dom';
import './SideMenu.scss';

export interface MenuItemSideMenu {
  name: string;
  url: string;
}

interface SideMenuProps {
  itemList: MenuItemSideMenu[];
}

const styles = (theme: Theme) => ({
  menuItem: {
    '&:focus': {
      backgroundColor: theme.palette.primary.main,
      '& $primary, & $icon': {
        color: theme.palette.common.white,
      },
    },
  },
  primary: {},
  icon: {},
});

const ListItemComposition: FunctionComponent<SideMenuProps> = (props) => {
  const { itemList } = props;

  return (
    <div>
      {props.children}
      <Paper>
        <MenuList>
          {itemList.map((item, index) => (
            <NavLink
              key={index}
              to={{
                pathname: item.url,
              }}
            >
              <MenuItem>{item.name}</MenuItem>
            </NavLink>
          ))}
          {/* {itemList.map((item, index) => 
        <MenuItemLink key={index} to={item.url}>
            {item.name}
        </MenuItemLink>
        )} */}
        </MenuList>
      </Paper>
    </div>
  );
};

export default withStyles(styles)(ListItemComposition);
