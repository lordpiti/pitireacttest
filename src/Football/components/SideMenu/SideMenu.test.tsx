import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SideMenu, { MenuItemSideMenu } from './SideMenu';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const menuData: MenuItemSideMenu[] = [
  { name: 'test1', url: 'test1' },
  { name: 'test2', url: 'test2' },
];

const history = createMemoryHistory();

// https://www.freecodecamp.org/news/8-simple-steps-to-start-testing-react-apps-using-react-testing-library-and-jest/

// testing with router:
// https://testing-library.com/docs/example-react-router

afterEach(cleanup);
describe('SideMenu', () => {
  it('renders correctly a sidemenu component with the right data', () => {
    const { asFragment } = render(
      <Router history={history}>
        <SideMenu itemList={menuData} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
