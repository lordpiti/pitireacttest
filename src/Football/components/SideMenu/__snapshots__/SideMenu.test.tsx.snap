// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideMenu renders correctly a sidemenu component with the right data 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <ul
        class="MuiList-root MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <a
          href="test1"
          tabindex="0"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            test1
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </a>
        <a
          href="test2"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            test2
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </a>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
